pkg/distributor/forwarding/forwarding.go

```diff
@@ -114,6 +184,10 @@ func (f *forwarder) start(ctx context.Context) error {
 func (f *forwarder) stop(_ error) error {
 	close(f.reqCh)
 	f.workerWg.Wait()
+
+	if err := services.StopAndAwaitTerminated(context.Background(), f.httpGrpcClientPool); err != nil {
+		return errors.Wrap(err, "failed to stop client pool")
```

Comment:
I think this could be confusing if it's not specific, because it could be misinterpreted as talking about the http clients.

```suggestion
		return errors.Wrap(err, "failed to stop grpc client pool")
```